### ISSUE TITLE
Run check_estimator on our estimators

### ISSUE THREAD

**Issue by TomAugspurger** [dask/dask-ml#70]
http://scikit-learn.org/stable/developers/contributing.html#rolling-your-own-estimator

It'd be nice if these passed.

**Comment by TomAugspurger** [comment]
Hmm, we'll have to think a bit more about this. It can involve some checks that might be expensive to do, e.g. length checking on a dask.dataframe.

**Comment by PeterDSteinberg** [comment]
Is the expensive part that length checking a dask.dataframe requires `.compute()`?  If there are some parts of the `check_estimator` logic we have to skip, let's at least keep the checks for `get_params()` and `set_params()` as methods.

**Comment by TomAugspurger** [comment]
Yeah things like `len` can be expensive for dask.DataFrames. Other examples would be checking for NaNs / infs: https://github.com/dask/dask-ml/pull/72/files#diff-1be11e310bf210381f8dd957dbc27074R173. Everything else we can do with just the metadata on the dask collection itself.

Though perhaps I'm being overly paranoid about the costs here. Either way, we will want to inform users about which algorithms make multiple passes over the data (like KMeans), so that they'll want to `.persist` the data before running `.fit`.
